### PR TITLE
Unit test

### PR DESCRIPTION
**Pull request by Dekz120**

### CODE REVIEW

File: elooplib/client.hpp [hunk 1]
```diff
@@ -8,21 +8,42 @@
...
 
+struct Command
+{
+    enum
+    {
+        Time,
+        Echo,
+        Compress,
+        Decompress,
+        NotExitsts
```
Comment (Yanpas) [inline elooplib/client.hpp]: Not exists. Better `Unknown`

File: elooplib/client.hpp [hunk 2]
```diff
@@ -8,21 +8,42 @@
...
+    } m_cmd;
+    std::string m_arg;
```
Comment (Yanpas) [inline elooplib/client.hpp]: Prefixing members with m is my team's codestyle, please fix

File: test/cmd_parser/command_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Command commandTest(std::string &cmd)
+{
+  Client client(-1);
+  client.updateRequestField(cmd.c_str(), cmd.size());
+  client.parseCommand();
+  return client.getCommand();
+}
+
+TEST_CASE("Command parser test")
```
Comment (Yanpas) [inline test/cmd_parser/command_parser.cpp]: Create two cases: Invalid commands and valid commands

File: test/file_cmd/file_cmd.cpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "../doctest.h"
+#include <filesystem>
+
+#include "archive.hpp"
+
+TEST_CASE("Compression test") {
```
Comment (Yanpas) [inline test/file_cmd/file_cmd.cpp:7]: Good test!

File: .github/workflows/make_check.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+name: C/C++ CI
+
```
Comment (Yanpas) [inline .github/workflows/make_check.yml:2]: You can also add building and running unit tests to Github-CI

File: test/Makefile [hunk 6]
```diff
@@ -0,0 +1,15 @@
+CXX = g++
+CXXFLAGS = -std=c++2a
```
Comment (Yanpas) [inline test/Makefile:2]: You can call this malefile from the top one using: `$(MAKE) -C test`

**Review by Yanpas — CHANGES_REQUESTED** [review]

File: elooplib/client.hpp (hunk 1 above)
Comment (Dekz120) [reply elooplib/client.hpp]: मेरी अंग्रेजी के लिए खेद है

File: test/cmd_parser/command_parser.cpp [hunk 7]
```diff
@@ -12,17 +12,25 @@ Command commandTest(std::string &cmd)
   return client.getCommand();
 }
 
-TEST_CASE("Command parser test")
+TEST_CASE("Vaild command parser test")
```
Comment (Yanpas) [inline test/cmd_parser/command_parser.cpp]: Valid

**Review by Yanpas — APPROVED** [review]

**Comment by Yanpas** [comment]
You have two identical CI jobs, don't you?

File: test/cmd_parser/command_parser.cpp (hunk 7 above)
Comment (Dekz120) [reply test/cmd_parser/command_parser.cpp]: Wild♂

**Comment by Yanpas** [comment]
Ping

**Comment by Dekz120** [comment]
> You have two identical CI jobs, don't you?

fixed in commit. but i have to resolve conlfict(in CI) before merge

**Review by Yanpas — APPROVED** [review]

**Pull request merged**